packages/babel-plugin-proposal-pipeline-operator/src/buildOptimizedSequenceExpression.js

```diff
@@ -0,0 +1,50 @@
+import { types as t } from "@babel/core";
+
+// tries to optimize sequence expressions in the format
+//   (a = b, ((c) => d + e)(a))
+// to
+//   (a = b, a + e)
+const buildOptimizedSequenceExpression = ({ assign, call, path }) => {
```

Comment:
(Note for reviewers: this function isn't new; it has been moved here from https://github.com/babel/babel/blob/b5febcf1f993cc3ec36fda7c7908181048a0069f/packages/babel-plugin-proposal-pipeline-operator/src/minimalVisitor.js)